### PR TITLE
Add test for nmt scripts

### PR DESCRIPTION
**Pull request by szhengac**
## Description ##
The current script is prone to error due to the api/base code change. To ensure the script can run properly without making error. This pr adds test for scripts in CI. 

A small translation dataset TEST is created for this purpose. The test code for scripts is added to scripts/tests/test_scripts.py.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Feature1, tests, (and when applicable, API doc)
- [x] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/api/data.rst [hunk 1]
```diff
@@ -102,7 +102,8 @@ We provide several standard datasets for machine translation.
 
 .. autosummary::
     :nosignatures:
-
+  
+    TEST
```
Comment (szha) [inline docs/api/data.rst]: no need to put in doc

File: gluonnlp/data/translation.py [hunk 2]
```diff
@@ -150,6 +150,49 @@ def tgt_vocab(self):
         return self._tgt_vocab
 
 
+@register(segment=['train', 'val', 'test'])
+class TEST(_TranslationDataset):
```
Comment (szha) [inline gluonnlp/data/translation.py]: leave this in scripts instead.

File: scripts/tests/test_scripts.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import subprocess
+from ..nmt.dataset import TEST
+
+
+def test_test():
```
Comment (szha) [inline scripts/tests/test_scripts.py]: haha let's pick a more informative name.

Comment (szhengac) [reply scripts/tests/test_scripts.py]: yeah, perhaps use test_translation_test.

**Comment by mli** [comment]
Job PR-234/19 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-234/19/index.html
